handlers.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"gopkg.in/mgo.v2/bson"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+func Auth(fn http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// TODO Handle auth.
+		// Only continue if user is authorized
+		fn(w, r)
+	}
+}
+
+func (sf *Sunfish) AddFile(w http.ResponseWriter, r *http.Request) {
+	// Handles a Post Request for a Sia file and saves it to the DB
+	var siafile Siafile
+	body, err := ioutil.ReadAll(io.LimitReader(r.Body, 1048576))
+
+	if err != nil {
+		panic(err)
+	}
```

Comment:
avoid `panic` if at all possible. logging the error and returning would be better in 99% of the cases I see here